src/ray/raylet/scheduling/cluster_resource_manager.cc

```diff
@@ -292,4 +292,22 @@ BundleLocationIndex &ClusterResourceManager::GetBundleLocationIndex() {
   return bundle_location_index_;
 }
 
+void ClusterResourceManager::UpdateNodeLabels(
+    const scheduling::NodeID &node_id,
+    const absl::flat_hash_map<std::string, std::string> &labels) {
+  if (labels.empty()) {
+    return;
+  }
+
+  auto it = nodes_.find(node_id);
+  if (it == nodes_.end()) {
+    NodeResources node_resources;
+    it = nodes_.emplace(node_id, node_resources).first;
+  }
+
+  for (const auto &[key, value] : labels) {
```

Comment:
We should clear existing labels and set to new ones?

Reply:
Same as above.
At present, there won't be any scenario where existing labels will contaminate the labels being analyzed. If such a scenario arises in the future, I will create a new interface called 'ResetNodeLabels()'.